src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/Actions/EliminationForms/FailedToQualifyForm.razor

```diff
@@ -28,62 +29,41 @@
 </MudButton>
 
 @code {
-    [Inject]
-    IParticipationBehind _participationBehind { get; set; } = default!;
     [Parameter]
     public FailedToQualify? FailedToQualify { get; set; }
-    [Parameter]
-    public EventCallback OnRestore { get; set; }
-    [Parameter]
-    public EventCallback OnEliminate { get; set; }
 
-    private IEnumerable<string>? FTQReason { get; set; }
+    private IEnumerable<FTQCodes> Codes { get; set; } = new List<FTQCodes>();
     public string? Reason { get; set; }
-    private bool _reasonInputState;
 
     protected override void OnParametersSet()
     {
-        FTQReason = ParseFTQCodesToStrings(FailedToQualify?.Codes);
-        Reason = FailedToQualify?.Complement;
-        _reasonInputState = FTQReason != null ? FTQReason.Contains("FTC") : false;
+       if(FailedToQualify != null)
+       {
+            Codes = FailedToQualify.Codes;
+            Reason = FailedToQualify.Complement;
+       }
     }
 
-    public async Task Eliminate()
+    public override async Task Eliminate()
     {
-        var tandemNumber = _participationBehind!.SelectedParticipation!.Tandem.Number;
-        List<FTQCodes> codes = new List<FTQCodes>();
-        foreach (var value in FTQReason)
+        var codes = new List<FTQCodes>();
+        foreach (var value in Codes)
         {
-            codes.Add((FTQCodes)Enum.Parse(typeof(FTQCodes), value));
+            codes.Add(value);
         }
         if (codes.Contains(FTQCodes.FTC))
         {
-            await _participationBehind.FailToCompleteLoop(tandemNumber, Reason, codes.ToArray());
+            await _participationBehind.FailToQualify(Reason, codes.ToArray());
         }
         else
         {
-            await _participationBehind.FailToQualify(tandemNumber, codes.ToArray());
+            await _participationBehind.FailToQualify(codes.ToArray());
         }
```

Comment:
We discussed that we don't need two separate methods for FTQ in Participation, right?

Reply:
Removed the calls from FailedToQualify.razor.
`public void FailToQualify(params FTQCodes[] codes)`
still in use at ParticipationFactory.cs